src/main/java/io/supertokens/inmemorydb/queries/TOTPQueries.java

```diff
@@ -139,16 +262,45 @@ public static int removeExpiredCodes(Start start)
         return update(start, QUERY, pst -> pst.setLong(1, System.currentTimeMillis()));
     }
 
-    public static int removeUsedCodes(Start start, String userId, String deviceName)
-            throws StorageQueryException, SQLException {
-        // Remove codes where userId matches the given userId
-        // ONLY required for inmemorydb, as it does not support foreign key constraints.
+    public static int deleteAllDevices_Transaction(Start start, Connection con, String userId)
+            throws SQLException, StorageQueryException {
+        String QUERY = "DELETE FROM " + Config.getConfig(start).getTotpUserDevicesTable()
+                + " WHERE user_id = ?;";
+        return update(con, QUERY, pst -> pst.setString(1, userId));
+    }
+
+    public static int deleteAllUsedCodes_Transaction(Start start, Connection con, String userId)
+            throws SQLException, StorageQueryException {
         String QUERY = "DELETE FROM " + Config.getConfig(start).getTotpUsedCodesTable()
-                + " WHERE user_id = ? AND device_name = ?;";
+                + " WHERE user_id = ?;";
+        return update(con, QUERY, pst -> pst.setString(1, userId));
+    }
 
-        return update(start, QUERY, pst -> {
-            pst.setString(1, userId);
-            pst.setString(2, deviceName);
+    public static int deleteUser_Transaction(Start start, Connection con, String userId)
+            throws SQLException, StorageQueryException {
+        String QUERY = "DELETE FROM " + Config.getConfig(start).getTotpUsersTable()
+                + " WHERE user_id = ?;";
+        return update(con, QUERY, pst -> pst.setString(1, userId));
+    }
+
+    public static void deleteAllDataForUser(Start start, String userId)
```

Comment:
see comment above - this function is no longer needed

Reply:
Removed.